src/nox/intercepting_io.cr

```diff
@@ -29,16 +29,20 @@ class Nox::InterceptingIO < IO
   end
 
   def write(slice : Bytes) : Nil
-    String.build(&.write(slice))
-      .lines
-      .each do |line|
-        result = String.build do |str|
-          str << @name.ljust(max_name_size + 1).sub(@name, @name.colorize(@color).to_s)
-          str << "| "
-          str << line
+    lines = String.build(&.write(slice)).split(/(?<=[\n\r])/)
+    lines.pop if lines.last.blank?
```

Comment:
The end of strings are likely to have newlines which produces a blank string as the last item in the list. Gotta remove it.